packages/grafana-ui/src/components/QueryEditor/InputGroup.tsx

```diff
@@ -1,16 +1,38 @@
-import { css } from '@emotion/css';
-import React from 'react';
+import { css, cx } from '@emotion/css';
+import React, { Children, cloneElement, isValidElement, ReactElement } from 'react';
 
 import { useStyles2 } from '../../themes';
 
-interface InputGroupProps {}
+interface InputGroupProps {
+  // we type the children props so we can test them later on
+  children: Array<ReactElement<{ className?: string; invalid?: unknown }>>;
+}
 
-export const InputGroup: React.FC<InputGroupProps> = ({ children }) => {
+export const InputGroup = ({ children }: InputGroupProps) => {
   const styles = useStyles2(getStyles);
 
-  return <div className={styles.root}>{children}</div>;
+  // Find children with an invalid prop, and set a class name to raise their z-index so all
+  // of the invalid border is visible
+  const modifiedChildren = Children.map(children, (child) => {
+    if (isValidElement(child) && child.props.invalid) {
+      return cloneElement(child, { className: cx(child.props.className, styles.invalidChild) });
+    }
```

Comment:
hmmm i think this needs more testing to make sure that `cloneElement()` doesn't do funky things to form controls when they change from valid -> invalid